src/big/big_decimal.cr

```diff
@@ -207,8 +207,9 @@ struct BigDecimal < Number
     check_division_by_zero other
     other.factor_powers_of_ten
 
-    scale = @scale - other.scale
+    scale = @scale >= other.scale ? @scale - other.scale : 0
     numerator, denominator = @value, other.@value
+    numerator *= TEN ** (other.scale - @scale) unless @scale >= other.scale
```

Comment:
It would be great to void the duplicate `@scale >= other.scale` comparison.

You can swap the first two lines and refactor the ternary operator and trailing `unless` into a single conditional.

Reply:
https://github.com/crystal-lang/crystal/pull/10628/commits/99f7402f872e8b0119a81ee32b623c58a12f059b uses a single conditional 😄